ssl/ssl_lib.cc

```diff
@@ -2611,10 +2607,6 @@ void SSL_set_quiet_shutdown(SSL *ssl, int mode) {
 int SSL_get_quiet_shutdown(const SSL *ssl) { return ssl->quiet_shutdown; }
 
 void SSL_set_shutdown(SSL *ssl, int mode) {
-  // It is an error to clear any bits that have already been set. (We can't try
-  // to get a second close_notify or send two.)
-  assert((SSL_get_shutdown(ssl) & mode) == SSL_get_shutdown(ssl));
-
```

Comment:
why remove this assertion?

Reply:
What does "messing up the state machine" mean for AWS-LC? Does OpenSSL avoid messing up their state machine without this assert some other way?

Reply:
I don't think OpenSSL has ever avoided this, in 1.1.1 this was even just a simple mode configuration: https://github.com/openssl/openssl/blob/OpenSSL_1_1_1-stable/ssl/ssl_lib.c#L4086-L4089
Based on the ancient BoringSSL commit (https://github.com/aws/aws-lc/commit/63006a913b772d48a364915c3155d51744a8f704), the concern was that setting the same flag twice in a row would clear the mode bits and mess up the state.

A lot has changed since then, this commit (https://github.com/aws/aws-lc/commit/fa214e4a18785a4281375b56c00732b470e9cd5b) consolidated the shutdown state so that flags can't be cleared like the original behavior.

Reply:
As sure good call, I'll update the documentation to say that it's ignored. I'm on the same page with you on this isn't much of a concern anymore.